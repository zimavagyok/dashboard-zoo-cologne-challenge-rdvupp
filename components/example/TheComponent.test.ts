import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import TheComponent from './TheComponent.vue'

describe('TheComponent', () => {
  test('mounts and is still the same', () => {
    expect(TheComponent).toBeTruthy()
    const wrapper = mount(TheComponent, {
      props: {
        modelValue: '',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('gets its starting input value from a prop', async () => {
    const modelValue = 'Thomas'
    const wrapper = mount(TheComponent, {
      props: {
        modelValue,
      },
    })

    const input = wrapper.get('input')
    expect(input.element.value).toBe(modelValue)
  })
})
