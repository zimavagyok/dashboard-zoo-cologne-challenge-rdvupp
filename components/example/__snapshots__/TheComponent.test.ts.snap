// Vitest Snapshot v1

exports[`TheComponent > mounts and is still the same 1`] = `
"<div class=\\"m-1 text-xs\\"><label class=\\"block\\">Super cool input:</label>
  <form><span class=\\"relative\\"><input type=\\"text\\" class=\\"border rounded p-1 outline-none\\" placeholder=\\"Your name...\\" autofocus=\\"\\"></span><button type=\\"submit\\" class=\\"bg-blue-200 rounded p-1\\"> Submit </button></form>
  <div class=\\"mt-2\\"><a style=\\"cursor: pointer;\\" to=\\"/\\"> Example link </a></div>
</div>"
`;
